lib/common/socket.c

```diff
@@ -682,6 +682,27 @@ int h2o_socket_get_ssl_cipher_bits(h2o_socket_t *sock)
     return sock->ssl != NULL ? SSL_get_cipher_bits(sock->ssl->ssl, NULL) : 0;
 }
 
+
+h2o_iovec_t h2o_socket_log_ssl_session_id(h2o_socket_t *sock, h2o_mem_pool_t *pool)
+{
+    h2o_iovec_t key;
+    unsigned id_len;
+    const unsigned char *id;
+
+    switch (sock->ssl->handshake.server.async_resumption.state) {
+        case ASYNC_RESUMPTION_STATE_COMPLETE:
+            id = SSL_SESSION_get_id(sock->ssl->ssl->session, &id_len);
```

Comment:
Please use `SSL_get_session` to avoid issues with OpenSSL 1.1.0 (IIRC direct access to members is forbidden in 1.1.x).

Also, I believe that there's a chance for the session pointer to be NULL, and think that doing a NULL check is necessary.